testcontainers/testcontainers-go
# fix: define a two-phase release process

- chore: improve dry_run log when hitting the Golang proxy
- chore: bump tc-go version in mkdocs
- docs: verify that the origin remote is set
- fix: define a two-phase release process



## What does this PR do?
This PR resolves the inconsistencies in the current release script, defining two separate phases for the release: pre-release and release.

The pre-release phase includes the modification of the files, based on the state of the version value in the `version.go` file. This version contains the next development version, and it's the value used to perform a release.
Therefore, that value drives the release process to set it as the version for all downstream modules and examples.

This pre-release phase will modify the files without creating a commit, so that an eventual release engineer is able to check that the git state is correct.

Once the modified files are correct, we can safely trigger the second phase with the `release.sh` script. This script does:

- commits the modified files.
- creates the git tags for the new release, in the core module and examples and modules.
- calculates the next development version (default a minor bump).
- modifies the `version.go` file with the next version, creating a commit for that.
- pushes the main branch and all tags to the remote **origin** repository.
- triggers the Golang proxy to fetch all modules in the project, based on the git tags.

As a consequence of these changes in the release scripts, the `RELEASE.md` has been updated.



## Why is it important?
We do not want to mess the state of a release. A wrong git tag in the repo and an eventual fetch of the Golang proxy will cache the tag forever, causing disastrous consequences to our users, as that cached version will be there forever. This happened with the `v0.20.0` release, were the script failed in the middle but created an inconsistent state of the git tags, so I manually performed a release deleting the upstream tags.






## How to test this PR

To perform a release in DRY_RUN mode:

1. change the `version.go` to a version of your interest: e.g. 1.0.0
2. run `DRY_RUN=false ./scripts/pre-release.sh`
  - the files will be modified, so all modules should contain the `v1.0.0` version in their go.mod files.
3. run `DRY_RUN=true ./scripts/release.sh`
  - a log output will show up with all the operations that are described in the release document, withouht changing the state of both the local and remote git repository.
